src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -4436,5 +4456,305 @@ void verify(CSharpCompilation compilation)
                 }
             }
         }
+
+        [Theory]
+        [CombinatorialData]
+        public void DuplicateDeclarations_01(
+            bool withCovariantReturns
+            )
+        {
+            var source = @"
+class Base
+{
+    public virtual void M(int x) => throw null; // 1
+    public virtual void M(int y) => throw null; // 2
+}
+
+class Derived : Base
+{
+    public override void M(int z) => throw null; // 3
+}
+";
+            var comp = CreateCompilation(withCovariantReturns: withCovariantReturns, source);
+            comp.VerifyDiagnostics(
+                // (5,25): error CS0111: Type 'Base' already defines a member called 'M' with the same parameter types
+                //     public virtual void M(int y) => throw null; // 2
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M").WithArguments("M", "Base").WithLocation(5, 25)
+                );
+
+            var member = (SourceMethodSymbol)comp.GlobalNamespace.GetMember("Derived.M");
+            bool useMethodImpl = member.RequiresExplicitOverride(out bool shouldWarn);
+            Assert.False(shouldWarn);
+            Assert.Equal(withCovariantReturns, useMethodImpl);
```

Comment:
>useMethodImpl [](start = 47, length = 13)

This should always be false

Reply:
For reasons previously explained, this value is intended and by design, and not affected by any changes in this PR.

---
In reply to: [452974765](https://github.com/dotnet/roslyn/pull/45849#discussion_r452974765) [](ancestors = 452974765)